modules/alookup/alookup_test.go

```diff
@@ -263,6 +263,94 @@ func TestDoLookup(t *testing.T) {
 	} else if res != nil {
 		t.Errorf("Expected no results, got %v", res)
 	}
+
+	// Case 11: Request for A giving AAAA should return empty response
+	mockResults["example.com"] = miekg.Result{
+		Answers: []interface{}{miekg.Answer{
+			Ttl:    3600,
+			Type:   "A",
+			Class:  "IN",
+			Name:   "example.com",
+			Answer: "2001:db8::4",
+		}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       miekg.DNSFlags{},
+	}
+
+	res, _, status, _ = l.DoLookup("example.com", "")
+	if status != zdns.STATUS_NO_ANSWER {
+		t.Errorf("Expected NO_ANSWER status, got %v", status)
+	} else if res != nil {
+		t.Errorf("Expected no results, got %v", res)
+	}
+
+	// Case 12: Loops in CNAME should terminate with no results
+	mockResults["cname.example.com"] = miekg.Result{
+		Answers: []interface{}{miekg.Answer{
+			Ttl:    3600,
+			Type:   "CNAME",
+			Class:  "IN",
+			Name:   "cname.example.com",
+			Answer: "example.com.",
+		}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       miekg.DNSFlags{},
+	}
+	mockResults["example.com"] = miekg.Result{
+		Answers: []interface{}{miekg.Answer{
+			Ttl:    3600,
+			Type:   "CNAME",
+			Class:  "IN",
+			Name:   "example.com",
+			Answer: "cname.example.com.",
+		}},
+		Additional:  nil,
+		Authorities: nil,
+		Protocol:    "",
+		Flags:       miekg.DNSFlags{},
+	}
+
+	res, _, status, _ = l.DoLookup("example.com", "")
+	if status != zdns.STATUS_NO_ANSWER {
+		t.Errorf("Expected NO_ANSWER status, got %v", status)
+	} else if res != nil {
+		t.Errorf("Expected no results, got %v", res)
+	}
+
+	// Case 13: Empty non-terminal domain should give no answer
```

Comment:
This should be no answer but also no error? Does STATUS_NO_ANSWER imply NOERROR?

Reply:
Removed `STATUS_NO_ANSWER` altogether because it is not a standard dns error code. In all cases where we were returning `STATUS_NO_ANSWER`, we are now returning `NOERROR`